app/screens/find_channels/unfiltered_list/index.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Database, Q} from '@nozbe/watermelondb';
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {MM_TABLES} from '@constants/database';
+import {queryAllMyChannel} from '@queries/servers/channel';
+import {queryJoinedTeams} from '@queries/servers/team';
+
+import UnfilteredList from './unfiltered_list';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+const {CHANNEL} = MM_TABLES.SERVER;
+const MAX_UNREAD_CHANNELS = 10;
+const MAX_CHANNELS = 20;
+
+const observeChannels = async (myChannels: MyChannelModel[]) => {
+    const channels = await Promise.all(myChannels.map((m) => m.channel.fetch()));
+    return channels.filter((c): c is ChannelModel => c !== null);
+};
+
+const queryMyChannelsByUnread = (database: Database, isUnread: boolean, sortBy: 'last_viewed_at' | 'last_post_at', take = MAX_CHANNELS, excludeIds?: string[]) => {
```

Comment:
Any chance these two functions will be reused? (and therefore sent to the queries file)